diff -uNdr freeorion.original/FreeOrion/client/human/CMakeLists.txt freeorion/FreeOrion/client/human/CMakeLists.txt
--- freeorion.original/FreeOrion/client/human/CMakeLists.txt	2011-07-26 21:42:11.000000000 +0400
+++ freeorion/FreeOrion/client/human/CMakeLists.txt	2011-07-27 01:45:03.016762058 +0400
@@ -5,16 +5,14 @@
 
 message("-- Configuring freeorion")
 
-if (WIN32)
-   find_package(OpenGL)
-   if (OPENGL_FOUND)
-       if (NOT OPENGL_GLU_FOUND)
-           message(FATAL_ERROR "OpenGL GLU library not found.")
-       endif ()
-       include_directories(${OPENGL_INCLUDE_DIR})
-   else ()
-       message(FATAL_ERROR "OpenGL library not found.")
+find_package(OpenGL)
+if (OPENGL_FOUND)
+   if (NOT OPENGL_GLU_FOUND)
+       message(FATAL_ERROR "OpenGL GLU library not found.")
    endif ()
+   include_directories(${OPENGL_INCLUDE_DIR})
+else ()
+   message(FATAL_ERROR "OpenGL library not found.")
 endif ()
 
 find_package(OpenAL)
@@ -220,13 +218,14 @@
     ${OGG_LIBRARY}
     ${VORBIS_LIBRARIES}
     ${BULLET_LIBRARIES}
+    ${OPENGL_LIBRARIES}
+    ${ZLIB_LIBRARY}
 )
 
 if (WIN32)
     link_directories(${BOOST_LIBRARYDIR})
     set(THIS_EXE_LINK_LIBS
         ${THIS_EXE_LINK_LIBS}
-        ${ZLIB_LIBRARY}
         ${OPENGL_gl_LIBRARY}
         ${OPENGL_glu_LIBRARY}
         jpeg.lib libpng.lib glew32.lib
diff -uNdr freeorion.original/FreeOrion/universe/Parser.h freeorion/FreeOrion/universe/Parser.h
--- freeorion.original/FreeOrion/universe/Parser.h	2011-07-26 21:42:00.000000000 +0400
+++ freeorion/FreeOrion/universe/Parser.h	2011-07-27 01:03:24.526671234 +0400
@@ -45,6 +45,7 @@
 ////////////////////////////////////////////////////////////
 struct Skip : boost::spirit::classic::grammar<Skip>
 {
+  Skip() { };
     template <class ScannerT>
     struct definition
     {
